src/driver/src/DefaultDriver.cpp

```diff
@@ -55,6 +66,103 @@ YAML::Node loadConfig(const std::string& source,
     }
 }
 
+YAML::Node parseExternal(const YAML::Node& external,
+                         YamlParameters& params,
+                         const fs::path& phaseConfig) {
+    int nodeSize = 1;
+
+    fs::path path(external["Path"].as<std::string>());
+    path = fs::absolute(phaseConfig / path);
+
+    if (!fs::is_regular_file(path)) {
+        auto os = std::ostringstream();
+        os << "Invalid path to external PhaseConfig: " << path
+           << ". Please ensure your workload file is placed in 'workloads/[subdirectory]/' and the "
+              "'Path' parameter is relative to the 'phase_configs/' directory";
+        throw InvalidConfigurationException(os.str());
+    }
+
+    auto replacement = loadConfig(path.string());
+
+    if (external["Parameters"]) {
+        nodeSize++;
+        auto newParams = external["Parameters"].as<YamlParameters>();
+        params.insert(newParams.begin(), newParams.end());
+    }
+
+    if (external["Key"]) {
+        nodeSize++;
+        const auto key = external["Key"].as<std::string>();
+        if (!replacement[key]) {
+            auto os = std::ostringstream();
```

Comment:
I've usually used `std::stringstream` since I think it's more universal but it looks like `std::ostringstream` may be slightly faster... No change required but I wonder if worth indicating the preference for `ostringstream` in the styleguide.

Reply:
I think there are legit use cases for all three. I copied this line from the description printing code and didn't look closely at `stringstream`. I'm personally not bothered by genny using both. It's like sometimes people use `\n` and some `endl`. But happy to put a note in the style guide if you'd like.